src/relayFeeCalculator/chain-queries/baseQuery.ts

```diff
@@ -126,7 +126,18 @@ export class QueryBase implements QueryInterface {
     deposit: Omit<Deposit, "messageHash">,
     relayer = toAddressType(getDefaultSimulatedRelayerAddress(deposit.destinationChainId), deposit.destinationChainId)
   ): Promise<PopulatedTransaction> {
-    return populateV3Relay(this.spokePool, deposit, relayer);
+    // TODO? We should confirm that these deposit fields were verified on ingestion, otherwise we will throw on casts here
+    // TODO? If anything, this `getUnsignedTxFromDeposit` function should already accept deposit & { ...fields: EvmAddress }-type argument
```

Comment:
have you tried changing the `deposit` type to `deposit & { ...fields: EvmAddress }`?